lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1885,6 +1913,38 @@ void TFDeabstraction::checkAndCanonicalizeAttributes() {
                    diag::tf_op_misuse, error)
             .highlight(loc.getSourceRange());
           isError = true;
+
+          // If we have more specific information about what went wrong, emit a
+          // note.
+
+
+          // This is a limitation of our current implementation, because we
+          // don't have a SIL instruction for Tensor ops, and have to use
+          // emitConstantInst which can fail.
+          if (it->second.getKind() != SymbolicValue::Unknown)
+            break;
+
+          std::pair<SILNode *, UnknownReason> unknown =
+            it->second.getUnknownValue();
+          auto badInst = dyn_cast<SILInstruction>(unknown.first);
+          if (!badInst)
+            break;
+
+          switch (unknown.second) {
+          case UnknownReason::Default:
+            error = "could not fold operation";
+            break;
+          case UnknownReason::TooManyInstructions:
+            // TODO: Should pop up a level of the stack trace.
+            error = "constant expression too large to evaluate";
```

Comment:
Since we aren't using the term "constant expression" in user-facing syntax, is it worth rewording this to something like:
"expression is too complex to evaluate at compile-time"